### ISSUE TITLE
Dynamic SQL Query for the things that arent SMO 

### ISSUE THREAD

**Issue by SQLDBAWithABeard** [dataplat/dbachecks#885]
How can we make one SQL query to gather information that we are not getting from SMO when the information that we need is dynamic dependant on the tags chosen (checks being run)?

Think memory dumps

**Comment by SQLDBAWithABeard** [comment]
@ClaudioESSilva @shaneis any ideas here ?

**Comment by shaneis** [comment]
Yeah, that works for me. 

Will we have to check for the existence of properties in the `[PSCustomObject]`?
Or do we go down the road of "if we're doing this test, we'll have the property"?

Not an issue either way, I suppose :shrug:

**Comment by SQLDBAWithABeard** [comment]
I think that will work

**Comment by ClaudioESSilva** [comment]
As far as I know, in case of any kind of error on the sql side (syntax, not existing object, etc) the whole query will fail so we won't have a recordset.

Maybe checking if we have 1 record is enough. Then we can go down the road.

**Comment by SQLDBAWithABeard** [comment]
that doesnt sound so good. Unless the reason that the query fails is that hte instance is not contactable

**Comment by ClaudioESSilva** [comment]
Agree. But, assuming that we are using the T-SQL queries that dbatools does, that should cover us from things like 'this dmv only appeared on SQL XXXX' and therefore return errors.
Regarding not contactable, sure, but we can confirm that is the current error.

**Comment by shaneis** [comment]
Oh yeah, I was more thinking do we want to be very defensive and go  "yes, I know we're in this code block but we should double-check if the returned object has the MemoryDump property"
Or do we go "Well, we're in this code block so we'd **_have_** to have the MemoryDump property"

Either way, I think Claudio has the best option
